src/channels.ts

```diff
@@ -0,0 +1,695 @@
+import Errors from './errors'
+
+import {
+  type Channel,
+  type GroupsPage,
+  type QueryParams,
+  type Permissions,
+  type Response,
+  type ChannelsPage,
+  type UsersPage
+} from './defs'
+export default class Channels {
+  // Channels API client
+  /**
+   * @class Channels -
+   * Channels API is used for managing Channels. It is used for creating new
+   * channels, retrieving them, updating them and disabling them
+   * @param {string} channels_url - URL to the Channels service
+   * @param {string} content_type - Content type for the requests which is an application
+   * json
+   * @param {string} channelsEndpoint - Endpoint for the channels' service.
+   * @returns {Object} -Channels object
+   *
+   */
+  private readonly usersUrl: URL
+  private readonly contentType: string
+  private readonly channelsEndpoint: string
+  private readonly channelError: Errors
+  private readonly thingsUrl: URL
+  private readonly channelUnshareEndpoint: string = 'unshare'
+  private readonly channelPermissionsEndpoint: string = 'permissions'
+  public constructor (usersUrl: string, thingsUrl: string) {
+    this.usersUrl = new URL(usersUrl)
+    this.thingsUrl = new URL(thingsUrl)
+    this.contentType = 'application/json'
+    this.channelsEndpoint = 'channels'
+    this.channelError = new Errors()
+  }
+
+  public async CreateChannel (
+    channel: Channel,
+    token: string
+  ): Promise<Channel> {
+    // Creates a new channel
+    /**
+     * @method Create - Creates new channels when provided with a channel object
+     * with viable fresh information and a valid token.
+     * @param {Object} channel - Channel Object with a name and id.
+     * @param {String} token - An access token that is valid.
+     * @returns {Object} - Channel object.
+     * @example
+     * const channel = {
+     * "name": "channelName",
+     * "description": "long channel description",
+     * "parent_id": "bb7edb32-2eac-4aad-aebe-ed96fe073879",
+     *  "metadata": {
+     *       "domain": "example.com"
+     *  },
+     * "status": "enabled",
+     * "owner_id": "bb7edb32-2eac-4aad-aebe-ed96fe073879"
+     * }
+     *
+     */
+    const options: RequestInit = {
+      method: 'post',
```

Comment:
Keep the method names in capital for all
```suggestion
      method: 'POST,
```